### ISSUE TITLE
go.mod causes go install to fail (must not contain directives ... interpreted differently than if it were the main module)

### ISSUE THREAD

**Issue by LaPingvino** [dolthub/dolt#6462]
```bash
$ go install github.com/dolthub/dolt/go/cmd/dolt@latest
go: downloading github.com/dolthub/dolt/go v0.40.4
go: github.com/dolthub/dolt/go/cmd/dolt@latest (in github.com/dolthub/dolt/go@v0.40.4):
        The go.mod file for the module providing named packages contains one or
        more replace directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module.
```

**Comment by timsehn** [comment]
Why do you want such an old version?

**Comment by LaPingvino** [comment]
I don't; when I use @latest it identifies this old one somehow? my guess is that the go directory has a superfluous go.mod file?

**Comment by timsehn** [comment]
So, I had to do a bunch of research to see what is going on here. 

First, `@latest` and `@main` tags like this are used to install go modules. Dolt is not a go module. It is a compiled binary you use like Git. So, this will not work. To install dolt, you git clone the git repo, `cd go/cmd/dolt`, and run `go install`. This puts a freshly built `dolt` binary in you `$GOPATH/bin`. Then if that is on your `$PATH` typing `dolt` in a shell and hitting enter runs the `dolt` program.

Second, the only reason this works at all is because at one point we published a tag call `go/v0.40.4`. This also got registered in the Go central module repository. None of us are sure how or why this happened. But, we're sure it never worked. You could never use Dolt like a module. That's why the command does anything at all. I deleted that tag but it's still registered in the Go module repository. We're not going to delete it. It's just there.

So, don't install dolt like this :-) Use the standard clone method described in our docs:

https://docs.dolthub.com/introduction/installation/source
